global-news-app/src/Hooks/SearchContext/SearchContext.tsx

```diff
@@ -0,0 +1,76 @@
+import { Selections, ValueSelection } from "Types";
+import { SelectLabels, SelectionsKeys } from "Types/enums";
+import React, { createContext, useState, useMemo } from "react";
+import useLocalStorage from "./useLocalStorage";
+
+export type SearchContextType = {
+  selections: Selections;
+  updateSelection: (key: string, value: ValueSelection) => void;
+};
+const initialState: SearchContextType = {
+  selections: {
+    articleType: {
+      value: SelectionsKeys.TOP_HEADLINES,
+      label: SelectLabels.TOP_HEADLINES,
+    },
+  },
+  updateSelection: () => {},
+};
+
+const SearchContext = createContext<SearchContextType>(initialState);
+
+export const SearchContextProvider = ({
+  children,
+}: React.PropsWithChildren<Record<string, unknown>>) => {
+  const [location] = useLocalStorage<any | null>("userLocation", null);
+
+  const [selections, setSelections] = useState<Selections>({
+    articleType: {
+      value: SelectionsKeys.TOP_HEADLINES,
+      label: SelectLabels.TOP_HEADLINES,
+    },
+    country: { value: location?.country_code, label: location?.country_name },
+  });
+
+  const updateSelection = (key: string, value: ValueSelection) => {
```

Comment:
as i remember correctly the SelectionsKeys is an enum so the type of the key should not be string